tardis/spectrum/formal_integral/source_function.py

```diff
@@ -0,0 +1,285 @@
+
+
+from tardis import constants as const
+
+from dataclasses import dataclass
+import numpy as np
+import pandas as pd
+import scipy.sparse as sp
+import scipy.sparse.linalg as linalg
+from astropy import units as u
+
+
+class SourceFunctionSolver:
+
+    def __init__(self, line_interaction_type):
+
+        self.line_interaction_type = line_interaction_type
+
+
+    def solve(self, sim_state, opacity_state, transport_state, atomic_data, levels):
```

Comment:
The `atomic_data` should be part of the init, because it does not change between runs of the solver. `levels` is also part of the atomic data, so that doesn't need to be a separate argument.